### PR TITLE
KAFKA-16795: Fix broken compatibility in kafka.tools.NoOpMessageFormatter, kafka.tools.DefaultMessageFormatter, and kafka.tools.LoggingMessageFormatter

### PR DESCRIPTION
**Pull request by brandboat**
related to https://issues.apache.org/jira/browse/KAFKA-16795

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@brandboat the solution of my previous comment (https://github.com/apache/kafka/pull/15991#discussion_r1605853497) was to parse the passed string and then convert it to current version. For example:
```java
    private String convertDeprecatedClass(String className) {
        if (className.equals("kafka.tools.DefaultMessageFormatter")) {
            System.err.println("WARNING: kafka.tools.DefaultMessageFormatter is deprecated and will be removed in the next major release. " +
                "Please use org.apache.kafka.tools.consumer.DefaultMessageFormatter instead");
            return DefaultMessageFormatter.class.getName();
        }
        return className;
    }
```

The benefit of this approach is that we don't need to add removed class back. WDYT?

**Comment by brandboat** [comment]
> The benefit of this approach is that we don't need to add removed class back. WDYT?

This one is way better, thanks !

**Review by chia7712 — APPROVED** [review]
@brandboat thanks for this great patch. LGTM

File: tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumerOptions.java [hunk 1]
```diff
@@ -349,6 +349,25 @@ private MessageFormatter buildFormatter() {
         return formatter;
     }
 
+    private String convertDeprecatedClass(String className) {
```
Comment (chia7712) [inline tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumerOptions.java]: nit: this can be static method

**Review by chia7712 — APPROVED** [review]

**Comment by chia7712** [comment]
@brandboat Thanks for your contribution. Those supports should be removed from 4.0.0. I have filed ticket (https://issues.apache.org/jira/browse/KAFKA-16830)  for you. Please feel free to reassign yourself if you have no free cycles.

**Pull request merged**